modules/apps/29-fee/ibc_middleware_test.go

```diff
@@ -621,10 +622,30 @@ func (suite *FeeTestSuite) TestOnAcknowledgementPacket() {
 				// retrieve the relayer acc balance and add the expected recv and ack fees
 				relayerAccBalance := sdk.NewCoins(suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), relayerAddr, sdk.DefaultBondDenom))
 				expPayeeAccBalance = relayerAccBalance.Add(packetFee.Fee.RecvFee...).Add(packetFee.Fee.AckFee...)
+			},
+			true,
+			func() {
+				// assert that the packet fees have been distributed
+				found := suite.chainA.GetSimApp().IBCFeeKeeper.HasFeesInEscrow(suite.chainA.GetContext(), packetID)
+				suite.Require().False(found)
 
-				// retrieve the refund acc balance and add the expected timeout fees
-				refundAccBalance := sdk.NewCoins(suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), refundAddr, sdk.DefaultBondDenom))
-				expRefundAccBalance = refundAccBalance.Add(packetFee.Fee.TimeoutFee...)
+				relayerAccBalance := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), relayerAddr, sdk.DefaultBondDenom)
+				suite.Require().Equal(expPayeeAccBalance, sdk.NewCoins(relayerAccBalance))
+
+				refundAccBalance := suite.chainA.GetSimApp().BankKeeper.GetBalance(suite.chainA.GetContext(), refundAddr, sdk.DefaultBondDenom)
+				suite.Require().Equal(initialRefundAccBal, sdk.NewCoins(refundAccBalance))
+			},
+		},
+		{
+			"success: some refunds",
+			func() {
+				// set timeout_fee > recv_fee + ack_fee
+				packetFee.Fee.TimeoutFee = sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdkmath.NewInt(400)))
```

Comment:
I know the tests do a lot of hardcoding with the numbers, but might be nice to do:
```go
packetFee.Fee.TimeoutFee = packet.Fee.Total().Add(sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdkmath.NewInt(100)))
```
or something like that